cli/src/pcluster/templates/imagebuilder_stack.py

```diff
@@ -133,16 +154,12 @@ def _add_cfn_parameters(self):
     def _add_resources(self):
         # Add default build tags information
         tags = copy.deepcopy(self.config.build.tags) or []
-        tags.append(BaseTag(key="pcluster_build_image", value=utils.get_installed_version()))
+        tags.append(BaseTag(key=PCLUSTER_IMAGE_NAME_TAG, value=self.image_name))
```

Comment:
I think the `PCLUSTER_IMAGE_NAME_TAG` tag should be added as last step of the parallelcluster_tag component execution, because if the image is created during the build stage (with the tag PCLUSTER_IMAGE_NAME_TAG) and then the test stage will fail, we'll end up in having an Image with PCLUSTER_IMAGE_NAME_TAG that is not valid.

And the PCLUSTER_IMAGE_NAME_TAG is used by the build-image to check if an Image is already present.

An alternative to add the PCLUSTER_IMAGE_NAME_TAG in the parallelcluster_tag component execution is that we use another tag to check if AMI has passed the test, and we add a check on this new tag on the build-image  command

Reply:
yes, I'd add this tag in tag component